src/main/java/gift/kakao/controller/KakaoController.java

```diff
@@ -0,0 +1,54 @@
+package gift.kakao.controller;
+
+import gift.kakao.dto.KakaoUserResponseDto;
+import gift.kakao.service.KakaoService;
+import gift.member.dto.MemberLoginResponseDto;
+import gift.member.service.MemberService;
+import java.net.URI;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/auth/kakao")
+public class KakaoController {
+    @Value("${kakao.client-id}")
+    private String clientId;
+
+    @Value("${kakao.redirect-uri}")
+    private String redirectUri;
+
+    private final KakaoService kakaoService;
+
+    private final MemberService memberService;
+
+    public KakaoController(KakaoService kakaoService, MemberService memberService) {
+        this.kakaoService = kakaoService;
+        this.memberService = memberService;
+    }
+
+    @GetMapping("/login")
+    public ResponseEntity<Void> authRedirectToKakao() {
+        String kakaoAuthUrl = "https://kauth.kakao.com/oauth/authorize" +
+            "?response_type=code" +
+            "&client_id=" + clientId +
+            "&redirect_uri=" + redirectUri;
```

Comment:
URI 생성은 KakaoService 에 위임하는 것이 어떨까요?
카카오 로그인에 대한 로직 관리는 서비스가 적절해 보이고, apiKey, redirectUrl 관리도 한 곳에서 할 수 있습니다!

추가로 이 uri 는 앱이 실행된 후에는 변하지 않을 것 같은데 매번 요청 때마다 생성하지 않아도 될 것 같아요!

Reply:
```suggestion
        URI uri = UriComponentsBuilder.fromUriString("https://kauth.kakao.com")
                .path("/oauth/authorize")
                .queryParam("response_type", "code")
                .queryParam("client_id", clientId)
                .queryParam("redirect_uri", redirectUri)
                .build().toUri();
```

URI를 만들 때는 builder를 활용하면 구조적으로 만들 수 있어 더 명확하고 안전하게 생성이 가능합니다!

Reply:
URI를 builder로 만들 수 있었네요! 감사합니다 😊